File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/Reports/ReportBuilderModel.ts
import Environment = require('App/Models/Environment');
import MetadataRepositoryModel = require('App/Models/Commons/MetadataRepositoryModel');
import ReportBuilderProjectionModel = require('App/Models/Reports/ReportBuilderProjectionModel');
import ReportBuilderConditionModel = require('App/Models/Reports/ReportBuilderConditionModel');
import ReportBuilderSortModel = require('App/Models/Reports/ReportBuilderSortModel');

class ReportBuilderModel {
    constructor() {
        this.Projections = [];
        this.Conditions = [];
        this.Sorts = [];
    }

    Entity: Environment;
    MetadataRepository: MetadataRepositoryModel;
    UDType: any;
    Projections: ReportBuilderProjectionModel[];
    Conditions: ReportBuilderConditionModel[];
    Sorts: ReportBuilderSortModel[];
}

export = ReportBuilderModel;

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Reports/Implementation/ReportBuilderModel.cs
using System;
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Model.Entities.Commons;

namespace VecompSoftware.DocSuiteWeb.Model.Reports
{
    public class ReportBuilderModel
    {
        #region [ Constructor ]
        public ReportBuilderModel()
        {
            Projections = new List<ReportBuilderProjectionModel>();
            Conditions = new List<ReportBuilderConditionModel>();
            SortItems = new List<ReportBuilderSortModel>();
        }
        #endregion

        #region [ Properties ]
        public DSWEnvironmentType Entity { get; set; }
        public Guid? MetadataRepositoryId { get; set; }
        public DSWEnvironmentType? DocumentUnit { get; set; }
        public ICollection<ReportBuilderProjectionModel> Projections { get; set; }
        public ICollection<ReportBuilderConditionModel> Conditions { get; set; }
        public ICollection<ReportBuilderSortModel> SortItems { get; set; }
        #endregion
    }
}
